lib/utils/resolveObject.js

```diff
@@ -100,9 +96,28 @@ function compile_value(value, stack) {
         // Recursive so we can compute multi-layer variables like a = b, b = c, so a = c
         if ( to_ret.indexOf('{') > -1 ) {
           var reference = to_ret.slice(1, -1);
-          if(stack.indexOf(reference)!==-1) {
-            stack.push(reference);
-            throw new Error('Variable reference for \'' + stack[0] + '\' resolves to circular definition cycle of \'' + reference + '\':  ' + stack.join(', ') );
+
+          // Compare to found circular references
+          var foundCircArr = Object.keys(foundCirc);
+          if (foundCircArr.indexOf(reference)!==-1) {
+            // If the current reference is a member of a circular reference, do nothing
+            // originally this was an error message, but that seems misleading because the circular reference itself is what needs to be fixed
+            // if(stack.filter(function(ref) {return foundCircArr.indexOf(ref)==-1}).length > 0) {
```

Comment:
extra commented code to be removed, or is part of the comment above?

Reply:
Eh, I thought it was explanatory since we have an if block with no actual code inside. Will delete the comment anyway.